plugins/modules/ndo_fex_device.py

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2025, Shreyas Srish (@shrsr) <ssrish@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: ndo_fex_device
+short_description: Manage FEX Devices on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage FEX Devices on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is supported on ND v3.2 (NDO v4.4) and later.
+author:
+- Shreyas Srish (@shrsr)
+options:
+  template:
+    description:
+    - The name of the template.
+    - The template must be a fabric resource policy template.
+    - This parameter or O(template_id) is required.
+    type: str
+  template_id:
+    description:
+    - The ID of the template.
+    - The template must be a fabric resource policy template.
+    - This parameter or O(template) is required.
+    type: str
+  name:
+    description:
+    - The name of the FEX Device.
+    type: str
+    aliases: [ fex_device ]
+  uuid:
+    description:
+    - The UUID of the FEX Device.
+    type: str
+    aliases: [ fex_device_uuid ]
+  description:
+    description:
+    - The description of the FEX Device.
+    type: str
+  fex_id:
+    description:
+    - The FEX ID.
+    - This parameter is required when O(state=present).
+    type: int
+  nodes:
+    description:
+    - The list of node IDs where the FEX Device will be deployed.
+    - Each element can either be a single node ID or a range of IDs in quotation marks ('').
+    - This parameter is required when O(state=present).
+    type: list
+    elements: str
+  interfaces:
+    description:
+    - The interface names connected to the FEX Device.
+    - This parameter is required when O(state=present).
+    type: list
+    elements: str
+  state:
+    description:
+    - Use C(absent) for removing.
+    - Use C(query) for listing an object or multiple objects.
+    - Use C(present) for creating or updating.
+    type: str
+    choices: [ absent, query, present ]
+    default: query
+notes:
+- The O(template) must exist before using this module in your playbook.
+  Use M(cisco.mso.ndo_template) to create the Fabric Policy template.
+seealso:
+- module: cisco.mso.ndo_template
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Create a FEX Device
+  cisco.mso.ndo_fex_device:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_test_template
+    name: ansible_test_fex_device
+    description: "FEX Device for Ansible"
+    fex_id: 102
+    nodes: [101, 102]
```

Comment:
```suggestion
    nodes: [101, 102, '103-105']
```
Should we include an example of a node range entry in the list?